port/unix/omrsock.c

```diff
@@ -560,15 +609,32 @@ omrsock_htonl(struct OMRPortLibrary *portLibrary, uint32_t val)
 int32_t
 omrsock_inet_pton(struct OMRPortLibrary *portLibrary, int32_t addrFamily, const char *addr, uint8_t *addrNetworkOrder)
 {
+	int32_t rc;
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
+	char * addrA2e;
+#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
+
 	if (NULL == addrNetworkOrder) {
 		return OMRPORT_ERROR_INVALID_ARGUMENTS;
 	}
 
-	if (1 != inet_pton(get_os_family(addrFamily), addr, addrNetworkOrder)) {
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
+	addrA2e = a2e(addr, strlen(addr));
+	rc = inet_pton(get_os_family(addrFamily), addrA2e, (void *)addrNetworkOrder);
+	free(addrA2e);
+#else /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
+	rc = inet_pton(get_os_family(addrFamily), addr, (void *)addrNetworkOrder);
+#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
```

Comment:
Maybe simpler as:

```c++
#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
	addr = a2e(addr, strlen(addr));
#endif

	rc = inet_pton(get_os_family(addrFamily), addr, (void *)result);

#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
	free(addr);
#endif
```

and then you can get rid of the `addrA2e` local as well.

Reply:
Sounds good, I will change it.

Reply:
@rwy0717 There is just one problem with changing it, addr is a const char * right now. I could change the addr argument to be a char *, but that may not be a good idea.

Is it okay if I do:
```
#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
	char *addrIn = a2e(addr, strlen(addr));
#else /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
	const char *addrIn = addr;
#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */

	rc = inet_pton(get_os_family(addrFamily), addrIn, (void *)addrNetworkOrder);

#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
	free(addrIn);
#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
```

Reply:
Oh you're right, I'm fine with the original then.